src/javascript/app/ng-wig/ngWigEditableDirective.js

```diff
@@ -14,7 +14,25 @@ angular.module('ngWig')
         ngModelController.$setViewValue($element.html());
       }
 
-      $element.bind('blur keyup change paste focus click', function() {
+      var eventsToBind = [
+        'blur',
+        'keyup',
+        'change',
+        'focus',
+        'click'
+      ];
+
+      if (angular.isFunction(scope.onPaste)) {
+        $element.on('paste', function(e) {
+          scope.onPaste(e, $element.html()).then(function(val) {
+            $element.html(val);
+          })
+        });
+      }else{
+        eventsToBind.push('paste');
+      }
+
+      $element.bind(eventsToBind.join(' '), function() {
```

Comment:
When put inside intercepting the paste event and altering content does not
work.

Stepan Suvorov <notifications@github.com> schrieb am So., 11. Okt. 2015
16:06:

> In src/javascript/app/ng-wig/ngWigEditableDirective.js
> <https://github.com/stevermeister/ngWig/pull/81#discussion_r41710805>:
>
> > +        'change',
> > +        'focus',
> > +        'click'
> > +      ];
> > +
> > +      if (angular.isFunction(scope.onPaste)) {
> > +        $element.on('paste', function(e) {
> > +          scope.onPaste(e, $element.html()).then(
>
> function(val) {
> > +            $element.html(val);
> > +          })
> > +        });
> > +      }else{
> > +        eventsToBind.push('paste');
> > +      }
> > +
> > +      $element.bind(eventsToBind.join(' '), function() {
>
> you can put 'paste' event inside.
> it should make code more readable,
> what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevermeister/ngWig/pull/81/files#r41710805>.
>